src/shiv/cli.py

```diff
@@ -163,10 +172,14 @@ def main(
                 if not Path(tmp_site_packages, "bin", console_script).exists():
                     sys.exit(NO_ENTRY_POINT.format(entry_point=console_script))
 
+        # Some projects need reproducible artifacts, so they can use SOURCE_DATE_EPOCH
+        # environment variable to specify the timestamps in the zipapp
+        timestamp = int(os.environ.get(SOURCE_DATE_EPOCH_ENV, time.time()))
+
         # create runtime environment metadata
         env = Environment(
-            built_at=str(datetime.now()),
-            build_id=str(uuid.uuid4()),
+            built_at=datetime.utcfromtimestamp(timestamp).strftime(BUILD_AT_TIMESTAMP_FORMAT),
+            build_id=None if reproducible_build_id else str(uuid.uuid4()),
```

Comment:
rather than `None`, let's make this `build_id=hashlib.sha256()`, and always use the hash of the contents as the build-id

So, create the seed hash object here, and later update it (in `_write_to_zip_app`), and put the call to hexdigest into `Environment.to_json`, wdyt?

Reply:
I think it is better to not set `build_id` at this point. It defaults to none and we overwrite it later when we write environment data to file.